Removed ts-ignore statements from codebase and fixed the tests in faulty files

yugal07 opened the pull request:
**What kind of change does this PR introduce?**

Bugfix

**Issue Number:**

Fixes #3228 

**Snapshots/Videos:**

N/A

**If relevant, did you update the documentation?**
Auto-generated docs

<!--Add link to Talawa-Docs.-->

**Summary**

Some files had ts-ignore statements which is considered to be bad practices, This PR fixes the faulty test files with proper tests and removes ts-ignore statements

**Does this PR introduce a breaking change?**

No

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Tests**
  - Enhanced type definitions for user roles and mock implementations in tests.
  - Updated mock return values to align with asynchronous behavior.
  - Expanded test coverage for various user scenarios, ensuring consistency and reliability in testing.
  - Refined import paths for improved clarity and structure.
  - Introduced a new test suite for the `Chat.updater` resolver, validating behavior under various conditions.
  - Added tests for unauthorized access and unauthenticated scenarios.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes reviewed (changes requested):
1. Ensure the test code coverage for your patch reaches as close to 100% as possible.
2. Please add a screen shot or validate via a reference to the codecov report to this PR
    - https://app.codecov.io/gh/PalisadoesFoundation/talawa-api/pull/3253?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation

yugal07 commented:
@coderabbitai full review

yugal07 commented:
![Screenshot from 2025-02-22 20-47-53](https://github.com/user-attachments/assets/e1e37732-1562-4a61-b359-b64ac66f6535)
![Screenshot from 2025-02-22 20-47-39](https://github.com/user-attachments/assets/19868e4b-bf62-4f3a-9572-80508c72f82f)

I have added additional test cases and both the files are now at 100% coverage. @palisadoes PTAL

palisadoes reviewed (changes requested):
These files need to be moved to these directories as `.test.` files like the rest of them. You are not following our convention stated clearly here:
- https://github.com/PalisadoesFoundation/talawa-api/blob/develop-postgres/src/README.md

Directories for the renamed `.test.` files 

1. test/graphql/types/Chat
2. test/graphql/types/Events

yugal07 commented on test/graphql/types/Chat/updatedAt.test.ts:
```diff
@@ -15,13 +44,14 @@ const mockParent = {
+
 const drizzleClientMock = {
 	query: {
 		usersTable: {
-			findFirst: vi.fn(),
+			findFirst: vi.fn().mockImplementation(() => Promise.resolve(undefined)),
 		},
 	},
-} as unknown as FastifyInstance["drizzleClient"];
+} as unknown as FastifyInstance["drizzleClient"] & MockDrizzleClient;
```
type of undefined is not assignable to this function. This will cause type errors

yugal07 commented on test/graphql/types/Event/updater.test.ts:
```diff
@@ -112,15 +135,40 @@
 		expect(result).toBeNull();
 	});
+
+	it("returns the current user if updaterId matches current user's id", async () => {
+		const mockCurrentUser: UserObject = {
+			id: "user_1",
+			role: "administrator",
+			organizationMembershipsWhereMember: [{ role: "administrator" }],
+		};
+
+		// Mock the event with updaterId matching the current user's id
+		const eventWithCurrentUserAsUpdater = {
+			...MockEvent,
+			updaterId: "user_1", // Same as mockCurrentUser.id
+		};
+
+		drizzleClientMock.query.usersTable.findFirst.mockResolvedValue(
+			mockCurrentUser,
+		);
+
+		const result = await resolveEventUpdater(
+			eventWithCurrentUserAsUpdater,
+			{},
+			authenticatedContext,
+		);
+		expect(result).toEqual(mockCurrentUser);
+	});
```
Incase a future contributor wants to change the values for a specific test, this would lead to unnecessary trouble.

yugal07 commented on test/graphql/types/Event/updater.test.ts:
```diff
@@ -1,7 +1,26 @@
+	id: string;
+	role: UserRole;
+	organizationMembershipsWhereMember: Array<{ role: UserRole }>;
+};
```
Done

palisadoes reviewed (changes requested):
The `src/graphql/types/Chat/updater.ts` file is only showing 13% test code coverage. Please add more comprehensive and valid tests to get this as close to 100% as possible.
1. Code coverage isn't reporting any significant change in the code health
    1. https://app.codecov.io/gh/PalisadoesFoundation/talawa-api/pull/3253?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation
1. Here is the coverage for the file in question. (Scroll down)
    1. https://app.codecov.io/gh/PalisadoesFoundation/talawa-api/blob/develop-postgres/src%2Fgraphql%2Ftypes%2FChat%2Fupdater.ts
    1. https://app.codecov.io/gh/PalisadoesFoundation/talawa-api/tree/develop-postgres?displayType=list
    ![image](https://github.com/user-attachments/assets/b5897b0f-7522-4aac-8499-d27faebb6ae1)

yugal07 commented:
The files i have worked on are src/graphql/types/Chat/updatedAt.ts and src/graphql/types/Event/updater.ts. Sure i will add tests for src/graphql/types/Chat/updater.ts, thanks for the review

yugal07 commented on test/graphql/types/Chat/updater.test.ts:
```diff
@@ -0,0 +1,282 @@
+
+		const result = await resolveUpdater(mockParent, {}, authenticatedContext);
+		expect(result).toEqual(mockUpdaterUser);
+	});
```
Done

yugal07 commented on test/graphql/types/Chat/updater.test.ts:
```diff
@@ -0,0 +1,270 @@
+
+	it("throws unauthenticated error when user is not found", async () => {
+		drizzleClientMock.query.usersTable.findFirst.mockResolvedValue(undefined);
+
```
Done

yugal07 commented:
@palisadoes, I have implemented all of the suggestions from coderabbit and added test cases for src/graphql/types/Chat/updater.ts as per your request. hence this PR now also solves issue #3058 . Let me know if theres need for more changes
